src/pint/bayesian.py

```diff
@@ -104,13 +104,17 @@ def _decide_likelihood_method(self):
                 ]
             )
             if not correlated_errors_present:
-                return "wls"
+                method = "wls"
             else:
                 raise NotImplementedError(
                     "GLS likelihood for correlated noise is not yet implemented."
                 )
                 # return "gls"
 
+        toa_type = "wb" if self.is_wideband else "nb"
+
+        return f"{method}-{toa_type}"

```

Comment:
Is having this method return a string which you then rely on the best practice?  Maybe instead this function can set some a property like `self.lnlikelihood` to the correct value (`self._wls_nb_lnlikelihood`, `self._wls_wb_lnlikelihood`)?  You could also add a `log.debug()` statement with the choice it's made.  This way you don't have to decide in one method and then decode that in another.

Reply:
I agree. I'll change this.

Reply:
done.